nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service-api/src/main/java/org/apache/nifi/elasticsearch/ElasticSearchClientService.java

```diff
@@ -191,9 +191,45 @@
      * @param query A JSON string reprensenting the query.
      * @param index The index to target. Optional.
      * @param type The type to target. Optional. Will not be used in future versions of Elasticsearch.
+     * @param requestParameters A collection of URL request parameters. Optional.
      * @return A SearchResponse object if successful.
      */
-    SearchResponse search(String query, String index, String type);
+    SearchResponse search(String query, String index, String type, Map<String, String> requestParameters);
+
+    /**
+     * Retrieve next page of results from a Scroll.
+     *
+     * @param scroll Scroll body containing scrollId and optional scroll (keep alive) retention period.
```

Comment:
To be consistent with the documentation on line 191, what about mentioning that the scroll body should be JSON?

Reply:
Done